### PR TITLE
PRMT-4587 - ehr-out - Integrate New Data Access Logic within Application

### PR DESCRIPTION
**Pull request by joefong-nhs**

### CODE REVIEW

File: src/__tests__/app.integration.test.js [hunk 1]
```diff
@@ -92,10 +87,10 @@ describe('Swagger Documentation', () => {
 });
 
 describe('GET /registration-requests/:conversationId', () => {
-  const conversationId = v4();
+  const conversationId = v4(); // OutboundConversationId
```
Comment (martin-nhs) [inline src/__tests__/app.integration.test.js:90]: Could we rename this variable please

```suggestion
  const outboundConversationId = v4();
```

File: src/constants/enums.js [hunk 2]
```diff
@@ -27,11 +32,31 @@ export const CoreStatus = {
...
+  INCORRECT_ODS_CODE: 'OUTBOUND:incorrect_ods_code',
+  MISSING_FROM_REPO: 'OUTBOUND:missing_from_repo',
+  EHR_DOWNLOAD_FAILED: 'OUTBOUND:ehr_download_failed',
+  CORE_SENDING_FAILED: 'OUTBOUND:core_sending_failed',
+  EHR_INTEGRATION_FAILED: 'OUTBOUND:ehr_integration_failed',
+
+  // Fragment level
+  DOWNLOAD_FAILED: 'OUTBOUND:download_failed',
+  SENDING_FAILED: 'OUTBOUND:sending_failed'
+};
```
Comment (martin-nhs) [inline src/constants/enums.js:58]: Let's make sure these are documented here https://nhse-dsic.atlassian.net/wiki/spaces/TW/pages/12509020254/ORC+-+Databases

File: src/models/acknowledgements.js [hunk 3]
```diff
@@ -51,6 +55,6 @@ const model = dataType => ({
     }
 });
 
-export default (sequelize, DataTypes) => {
-    return sequelize.define(modelName, model(DataTypes), getParameters(tableName));
-};
+// export default (sequelize, DataTypes) => {
+//     return sequelize.define(modelName, model(DataTypes), getParameters(tableName));
+// };
```
Comment (martin-nhs) [inline src/models/acknowledgements.js:60]: Worth completely removing this?

File: src/services/handler/__test__/continue-message-handler.test.js [hunk 4]
```diff
@@ -1,201 +1,231 @@
...
-  });
-});
+import {
+  patientAndPracticeOdsCodesMatch,
+  updateConversationStatus
+} from '../../transfer/transfer-out-util';
+// import {
+//   getNhsNumberByOutboundConversationId,
+//   getOutboundConversationById
+// } from "../../database/registration-request-repository";
```
Comment (martin-nhs) [inline src/services/handler/__test__/continue-message-handler.test.js:8]: Can this be removed?

File: src/services/handler/__test__/continue-message-handler.test.js [hunk 5]
```diff
@@ -1,201 +1,231 @@
...
+  getNhsNumberByOutboundConversationId,
+  getOutboundConversationById
+} from '../../database/dynamodb/outbound-conversation-repository';
+
+// Mocking
+jest.mock('../../../middleware/logging');
+jest.mock('../../transfer/transfer-out-fragments');
+jest.mock('../../transfer/transfer-out-util');
+jest.mock('../../database/dynamodb/outbound-conversation-repository');
+// jest.mock('../../database/registration-request-repository');
```
Comment (martin-nhs) [inline src/services/handler/__test__/continue-message-handler.test.js:33]: Can this be removed please?

File: src/constants/enums.js (hunk 2 above)
Comment (MohammadIqbalAD-NHS) [reply src/constants/enums.js:58]: Updated documentation!

File: src/services/handler/__test__/continue-message-handler.test.js (hunk 5 above)
Comment (MohammadIqbalAD-NHS) [reply src/services/handler/__test__/continue-message-handler.test.js:33]: Done

File: src/services/handler/__test__/continue-message-handler.test.js (hunk 4 above)
Comment (MohammadIqbalAD-NHS) [reply src/services/handler/__test__/continue-message-handler.test.js:8]: Done

**Review by MohammadIqbalAD-NHS — APPROVED** [review]

**Pull request merged**